lib/AST/ASTPrinter.cpp

```diff
@@ -1722,6 +1720,8 @@ void PrintAST::printAccessors(AbstractStorageDecl *ASD) {
     case WriteImplKind::InheritedWithObservers:
       PrintAccessor(ASD->getWillSetFunc());
       PrintAccessor(ASD->getDidSetFunc());
+      if (ASD->supportsMutation())
```

Comment:
It seems like we should *either* print willSet/didSet *or* print the setter. Maybe that's *another* setting we should be checking.

Reply:
I think printing didSet/willSet is probably easier. Otherwise you need a way to print a _stored_ property that has a setter, which we don’t have syntax for.

Reply:
It wouldn't be correct to expose that information. Harlan and I have been talking about various ways to indicate that a property is stored despite having a custom setter.

Reply:
I guess we could do it without another check by having PrintAccessor return whether the decl was present but skipped vs. (absent || printed).